environment/src/main/java/com/sequenceiq/environment/environment/flow/upgrade/ccm/handler/UpgradeCcmOnDatalakeHandler.java

```diff
@@ -35,7 +45,27 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
         LOGGER.debug("In UpgradeCcmOnDatalakeHandler.accept");
         EnvironmentDto environmentDto = environmentDtoEvent.getData();
         try {
-            // TODO action here
+            LOGGER.debug("Calling Upgrade CCM endpoint on data lake service");
+            SdxCcmUpgradeResponse ccmUpgradeResponse = sdxService.upgradeCcm(environmentDto.getResourceCrn());
+            switch (ccmUpgradeResponse.getResponseType()) {
+                case ERROR:
+                    String message = String.format("Upgrade CCM returned with reason: %s", ccmUpgradeResponse.getReason());
+                    LOGGER.warn(message);
+                    sendFailedEvent(environmentDtoEvent, environmentDto, new OperationException(message));
+                    return;
+                case TRIGGERED:
+                    LOGGER.debug("Waiting for data lake Upgrade CCM flow to finish.");
+                    pollerService.waitForUpgradeCcm(environmentDto.getId(), ccmUpgradeResponse.getResourceCrn());
```

Comment:
Out of curiosity: The FreeIPA upgrade step seems to be structured differently. `UpgradeCcmOnFreeIpaHandler` invokes `freeIpaPollerService.waitForCcmUpgrade()` that in turn calls `freeIpaService.upgradeCcm()`. Here the `sdxService.upgradeCcm()` invocation is in the handler itself. Is there any reason to follow a different pattern here?

Reply:
diversity...